framework/math/quadratures/angular/product_quadrature.cc

```diff
@@ -164,49 +161,70 @@ AngularQuadratureProdGL::AngularQuadratureProdGL(int Nphemi, bool verbose) : Pro
   AssembleCosines(azimu_ang, polar_ang, weights, verbose);
 }
 
-AngularQuadratureProdGLC::AngularQuadratureProdGLC(int Na, int Np, bool verbose)
+GLCProductQuadrature2DXY::GLCProductQuadrature2DXY(int Npolar, int Nazimuthal, bool verbose)
+  : ProductQuadrature(2)
 {
-  GaussLegendreQuadrature gl_polar(Np * 2);
-  GaussChebyshevQuadrature gc_azimu(Na * 4);
+  if (Npolar % 2 != 0)
+    throw std::invalid_argument("GLCProductQuadraturee2DXY: Npolar must be even.");
+
+  if (Nazimuthal % 4 != 0)
+    throw std::invalid_argument("GLCProductQuadraturee2DXY: Nazimuthal must be a multiple of 4.");
+
+  GaussLegendreQuadrature gl_polar(Npolar);
+  GaussChebyshevQuadrature gc_azimu(Nazimuthal);
 
   // Create azimuthal angles
   azimu_ang.clear();
-  for (unsigned int i = 0; i < (Na * 4); ++i)
-    azimu_ang.emplace_back(M_PI * (2 * (i + 1) - 1) / (Na * 4));
+  for (auto i = 0; i < Nazimuthal; ++i)
+    azimu_ang.emplace_back(M_PI * (2 * (i + 1) - 1) / Nazimuthal);
 
   // Create polar angles
   polar_ang.clear();
-  for (unsigned int j = 0; j < (Np * 2); ++j)
+  for (auto j = 0; j < Npolar; ++j)
     polar_ang.emplace_back(M_PI - acos(gl_polar.qpoints[j][0]));
 
   // Create combined weights
   std::vector<double> weights;
-  for (unsigned int i = 0; i < azimu_ang.size(); ++i)
-    for (unsigned int j = 0; j < polar_ang.size(); ++j)
+  for (auto i = 0; i < azimu_ang.size(); ++i)
+    for (auto j = 0; j < polar_ang.size(); ++j)
       weights.emplace_back(2 * gc_azimu.weights[i] * gl_polar.weights[j]);
 
   // Initialize
   AssembleCosines(azimu_ang, polar_ang, weights, verbose);
+
+  OptimizeForPolarSymmetry(4.0 * M_PI);
```

Comment:
what is the 4pi for? Don't we want, by default to normlaize to sum of weights = 1 no matter what?

Reply:
For this commit, I left the default as it was. There are so many other changes that I was purposefully trying to not have to modify any of the outputs. I have a second PR after this one is merged that makes that change since it's likely to affect the results of some tests.

Reply:
we have to be careful. a test that has a scalar flux as qoi will NOT change its value. so, if in the next PR, you need to re-gold a results, we have to scrutinize that